pkg/operator/controller/gatewayclass/istio.go

```diff
@@ -0,0 +1,209 @@
+			Namespace:       name.Namespace,
+			Name:            name.Name,
+			OwnerReferences: []metav1.OwnerReference{ownerRef},
+		},
+		Spec: sailv1.IstioSpec{
+			Namespace: controller.DefaultOperandNamespace,
+			UpdateStrategy: &sailv1.IstioUpdateStrategy{
+				Type: sailv1.UpdateStrategyTypeInPlace,
+			},
+			Values: &sailv1.Values{
+				Global: &sailv1.GlobalConfig{
+					IstioNamespace: ptr.To(controller.DefaultOperandNamespace),
+					// Logging: &sailv1.GlobalLoggingConfig{
+					// 	Level: ptr.To("all:debug"),
+					// },
+				},
+				Pilot: &sailv1.PilotConfig{
+					Cni: &sailv1.CNIUsageConfig{
+						Enabled: ptr.To(false),
+					},
+					Enabled:            ptr.To(true),
+					Env:                pilotContainerEnv,
+					ExtraContainerArgs: []string{},
+				},
+				SidecarInjectorWebhook: &sailv1.SidecarInjectorConfig{
+					EnableNamespacesByDefault: ptr.To(false),
+				},
+				MeshConfig: &sailv1.MeshConfig{
+					AccessLogFile: ptr.To("/dev/stdout"),
+					DefaultConfig: &sailv1.MeshConfigProxyConfig{
+						ProxyHeaders: &sailv1.ProxyConfigProxyHeaders{
+							Server: &sailv1.ProxyConfigProxyHeadersServer{
+								Disabled: ptr.To(true),
+							},
+							EnvoyDebugHeaders: &sailv1.ProxyConfigProxyHeadersEnvoyDebugHeaders{
+								Disabled: ptr.To(true),
+							},
+							MetadataExchangeHeaders: &sailv1.ProxyConfigProxyHeadersMetadataExchangeHeaders{
+								Mode: sailv1.ProxyConfigProxyHeadersMetadataExchangeModeInMesh,
+							},
+						},
+					},
+					IngressControllerMode: sailv1.MeshConfigIngressControllerModeOff,
+				},
+			},
+			Version: "v1.24.3",
+		},
+	}
+	return &istio, nil
+}
+
+// currentIstio returns the current istio CR.
+func (r *reconciler) currentIstio(ctx context.Context, name types.NamespacedName) (bool, *sailv1.Istio, error) {
+	var istio sailv1.Istio
+	if err := r.cache.Get(ctx, name, &istio); err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil, nil
+		}
+		return false, nil, fmt.Errorf("failed to get istio %s: %w", name, err)
+	}
+	return true, &istio, nil
+}
+
+// createIstio creates an Istio CR.
+func (r *reconciler) createIstio(ctx context.Context, istio *sailv1.Istio) error {
+	if err := r.client.Create(ctx, istio); err != nil {
+		return fmt.Errorf("failed to create istio %s/%s: %w", istio.Namespace, istio.Name, err)
+	}
+	log.Info("created istio", "namespace", istio.Namespace, "name", istio.Name)
+	return nil
+}
+
+// istioCmpOpts is an array of options for go-cmp to use when comparing Istio
+// CRs.
+var istioCmpOpts = []cmp.Option{
+	cmpopts.EquateEmpty(),
+}
+
+// updateIstio updates an Istio CR.
+func (r *reconciler) updateIstio(ctx context.Context, current, desired *sailv1.Istio) (bool, error) {
+	changed, updated := istioChanged(current, desired)
+	if !changed {
+		return false, nil
+	}
+
+	// Diff before updating because the client may mutate the object.
+	// Only diff spec because status may include unexported fields that
+	// cause go-cmp to panic.
```

Comment:
I wonder if there are labels/annotations that don't want to compare as well, e.g. if the istiod/proxy reconciles add labels/annotations that we are supposed to ignore?